app/scripts/migrations/135.ts

```diff
@@ -1,41 +1,58 @@
-import { hasProperty, isObject, NonEmptyArray, Json } from '@metamask/utils';
+import { hasProperty, isObject } from '@metamask/utils';
+import type { CaipChainId, CaipAccountId, Json } from '@metamask/utils';
 import { cloneDeep } from 'lodash';
+import type {
+  Caveat,
+  PermissionConstraint,
+  ValidPermission,
+} from '@metamask/permission-controller';
 
-type CaveatConstraint = {
-  type: string;
-  value: Json;
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
 };
 
-type PermissionConstraint = {
-  parentCapability: string;
-  caveats: null | NonEmptyArray<CaveatConstraint>;
-};
+export const version = 135;
 
-const PermissionNames = {
-  eth_accounts: 'eth_accounts',
-  permittedChains: 'endowment:permitted-chains',
+// In-lined from @metamask/multichain
+const Caip25CaveatType = 'authorizedScopes';
+const Caip25EndowmentPermissionName = 'endowment:caip25';
+
+type InternalScopeObject = {
+  accounts: CaipAccountId[];
 };
 
-const BUILT_IN_NETWORKS = {
-  goerli: '0x5',
-  sepolia: '0xaa36a7',
-  mainnet: '0x1',
-  'linea-goerli': '0xe704',
-  'linea-sepolia': '0xe705',
-  'linea-mainnet': '0xe708',
+type InternalScopesObject = Record<CaipChainId, InternalScopeObject>;
+
+type Caip25CaveatValue = {
+  requiredScopes: InternalScopesObject;
+  optionalScopes: InternalScopesObject;
+  sessionProperties?: Record<string, Json>;
+  isMultichainOrigin: boolean;
 };
 
-const Caip25CaveatType = 'authorizedScopes';
-const Caip25EndowmentPermissionName = 'endowment:caip25';
+// Locally defined types
+type Caip25Caveat = Caveat<typeof Caip25CaveatType, Caip25CaveatValue>;
+type Caip25Permission = ValidPermission<
+  typeof Caip25EndowmentPermissionName,
+  Caip25Caveat
+>;
 
-const snapsPrefixes = ['npm:', 'local:'] as const;
+const PermissionNames = {
+  eth_accounts: 'eth_accounts',
+  permittedChains: 'endowment:permitted-chains',
+} as const;
 
-type VersionedData = {
-  meta: { version: number };
-  data: Record<string, unknown>;
-};
+const BUILT_IN_NETWORKS: ReadonlyMap<string, string> = new Map([
```

Comment:
thoughts on using `Hex` here to the value type?

Reply:
`goerli` still exists in state by this migration?

Reply:
Hex works for the values but not the keys

Reply:
Or you mean `goerli` exists as a value?